code/controllers/CMSMain.php

```diff
@@ -1316,7 +1316,6 @@ public function duplicate($request) {
 		
 		if(($id = $this->urlParams['ID']) && is_numeric($id)) {
 			$page = DataObject::get_by_id("SiteTree", $id);
-			if($page && (!$page->canEdit() || !$page->canCreate())) return Security::permissionFailure($this);
```

Comment:
I think this should still include a canView check and a canCreate check. If not then a user could potentially navigate to the /duplicate URL and duplicate pages they don't have permission to see or create.